cmd/auth/login.go

```diff
@@ -20,7 +37,33 @@ var loginCmd = &cobra.Command{
 		defer perisistentAuth.Close()
 		ctx, cancel := context.WithTimeout(cmd.Context(), loginTimeout)
 		defer cancel()
-		return perisistentAuth.Challenge(ctx)
+
+		var profileName string
+		profileFlag := cmd.Flag("profile")
+		if profileFlag != nil && profileFlag.Value.String() != "" {
+			profileName = profileFlag.Value.String()
+		} else {
+			profiles, err := getProfileNames()
+			if err != nil {
+				return err
+			}
+			profile, err := cmdio.SelectWithAdd(ctx, profiles, "~/.databrickscfg profile", "Add a new profile")
+			if err != nil {
+				return err
+			}
+			profileName = profile
+		}
+		err := perisistentAuth.Challenge(ctx)
+		if err != nil {
+			return err
+		}
+
+		return databrickscfg.SaveToProfile(ctx, &config.Config{
+			Host:      perisistentAuth.Host,
+			AccountID: perisistentAuth.AccountID,
+			AuthType:  "databricks-cli",
+			Profile:   profileName,
+		})
```

Comment:
Please add some kind of confirmation at the end, saying the profile called "zzz" was saved.